redis/lolwut.cc

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#define _USE_MATH_DEFINES
+#include <cmath>
+#include <random>
+#include <seastar/core/sstring.hh>
+#include "redis/lolwut.hh"
+#include "redis/version.hh"
+
+using namespace seastar;
+
+namespace redis {
+
+class canvas {
+    const int _width, _height;
+    char *_pixels;
```

Comment:
Better to use `std::unique_ptr<char[]> _pixels;`:

 - you get the destructor defined
 - with the current canvas class, it it is copied, then the destructor will run on both copies resulting in double-free.